### PR TITLE
Completed POST requests

### PR DESCRIPTION
**Pull request by cjhbh3**
tl;dr Completed Step 4 Post Request of Week 3 starter project

Adds Servlet 3: a form that prompts for a text entry (for correct usage, a comma separated list) and has two toggle options. The toggles options which change the text entered by making the words uppercase or sorted alphabetically. Once, those changes have been made (if needed), the entered words are written onto /data. The user is then redirected back to index.html, where they can use Servlet 2 to see what they have done.

Additional changes:
- In DataServlet.java, convertToJson now utilizes Gson to construct JSON strings (instead of the former manual building of them). This was done as an easier and more efficient form of handling user inputted comments.
- In pom.xml, a dependency for the Gson library was added to allow for the use of Gson.
- In index.html, some visual changes have been made to make the site look better.

Notes: will remember to remove the commented out parts not being used.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -41,17 +38,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(json);
   }
 
+  @Override 
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Receive info from form
+    // String firstName = getParameter(request,"fname","");
+    // String lastName = getParameter(request,"lname","");
```
Comment (ptai7) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:45]: Remove unused comments.

**Review by ptai7 — APPROVED** [review]
Thanks, please re-deploy your website when submitting pull requests so that it can help us review the code/website display.

**Comment by cjhbh3** [comment]
> Thanks, please re-deploy your website when submitting pull requests so that it can help us review the code/website display.
I will go ahead and deploy right now

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -41,17 +38,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    boolean sort = Boolean.parseBoolean(getParameter(request,"sort","false"));
+
+    // Apply Settings
+    if (upperCase) {
+      information = information.toUpperCase();
+    }
+
+    // Break up comma split list into words
+    String[] words = information.split("\\s*, \\s*");
+
```
Comment (chrisc94) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:57]: I may have forgotten my Java but do you need all the regex there? I believe .split(",") is sufficient.

https://www.geeksforgeeks.org/split-string-java-examples/

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -41,17 +38,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    if (sort) {
+      Arrays.sort(words);
+    }
+
+    // Add words to our message ArrayList
+    for (String word : words) {
```
Comment (chrisc94) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:63]: you can use Arrays.asList(array)

https://www.tutorialspoint.com/Conversion-of-Array-To-ArrayList-in-Java

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (cjhbh3) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:57]: I believe you can. Will change that next update

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (cjhbh3) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:63]: I actually did not know that. I will keep that in mind.
